### PR TITLE
Miscellaneous improvements and clean-up.

### PR DESCRIPTION
**Pull request by leo-stan**

### CODE REVIEW

File: cpp/include/copc-lib/copc/extents.hpp [hunk 1]
```diff
@@ -129,43 +130,49 @@ class CopcExtents
...
+    }
+    void EdgeOfFlightLine(std::shared_ptr<CopcExtent> edge_of_flight_line)
+    {
+        extents_[5] = std::move(edge_of_flight_line);
+    }
+    void Classification(std::shared_ptr<CopcExtent> classification) { extents_[6] = std::move(classification); }
+    void UserData(std::shared_ptr<CopcExtent> user_data) { extents_[7] = std::move(user_data); }
+    void ScanAngle(std::shared_ptr<CopcExtent> scan_angle) { extents_[8] = std::move(scan_angle); }
+    void PointSourceId(std::shared_ptr<CopcExtent> point_source_id) { extents_[9] = std::move(point_source_id); }
+    void GpsTime(std::shared_ptr<CopcExtent> gps_time) { extents_[10] = std::move(gps_time); }
```
Comment (CCInc) [inline cpp/include/copc-lib/copc/extents.hpp:149]: what does all this do?

File: cpp/include/copc-lib/las/header.hpp [hunk 2]
```diff
@@ -101,9 +102,18 @@ class LasHeader
...
-    double ApplyInverseScaleZ(double scaled_value) const { return (scaled_value - offset_.z) / scale_.z; }
+    int ApplyInverseScaleX(double scaled_value) const
+    {
+        return static_cast<int>((scaled_value - offset_.x) / scale_.x);
+    }
+    int ApplyInverseScaleY(double scaled_value) const
+    {
+        return static_cast<int>((scaled_value - offset_.y) / scale_.y);
+    }
+    int ApplyInverseScaleZ(double scaled_value) const
```
Comment (CCInc) [inline cpp/include/copc-lib/las/header.hpp:113]: good catch, this should probably be a int64_t though.

File: cpp/src/io/copc_reader.cpp [hunk 3]
```diff
@@ -62,7 +62,7 @@ std::map<uint64_t, las::VlrHeader> Reader::ReadVlrHeaders()
     }
 
     // Move stream to beginning of EVLRs
-    in_stream_->seekg(reader_->header().evlr_offset);
+    in_stream_->seekg(static_cast<long>(reader_->header().evlr_offset));
```
Comment (CCInc) [inline cpp/src/io/copc_reader.cpp:65]: undo

File: cpp/include/copc-lib/copc/extents.hpp (hunk 1 above)
Comment (leo-stan) [reply cpp/include/copc-lib/copc/extents.hpp:149]: Another Clang modernization https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference

File: cpp/include/copc-lib/las/header.hpp (hunk 2 above)
Comment (leo-stan) [reply cpp/include/copc-lib/las/header.hpp]: What's the reasoning to force it to int64?

File: example/example-writer.cpp [hunk 4]
```diff
@@ -220,9 +220,9 @@ las::Points RandomPoints(const VoxelKey &key, const las::LasHeader &header, int
         // The use of las::Point constructor is strongly discouraged, instead use las::Points::CreatePoint
         auto point = points.CreatePoint();
         // point has getters/setters for all attributes
-        point->UnscaledX(rand_x(gen));
-        point->UnscaledY(rand_y(gen));
-        point->UnscaledZ(rand_z(gen));
+        point->UnscaledX(static_cast<int>(rand_x(gen)));
+        point->UnscaledY(static_cast<int>(rand_y(gen)));
+        point->UnscaledZ(static_cast<int>(rand_z(gen)));
```
Comment (CCInc) [inline example/example-writer.cpp:225]: shouldn't need these casts

**Review by CCInc — APPROVED** [review]

**Pull request merged**